ui/hooks/useAccountTotalFiatBalance.js

```diff
@@ -0,0 +1,92 @@
+import { shallowEqual, useSelector } from 'react-redux';
+import {
+  getAllTokens,
+  getCurrentChainId,
+  getCurrentCurrency,
+  getMetaMaskCachedBalances,
+  getTokenExchangeRates,
+} from '../selectors';
+import {
+  getValueFromWeiHex,
+  sumDecimals,
+} from '../../shared/modules/conversion.utils';
+import { getConversionRate } from '../ducks/metamask/metamask';
+import { formatCurrency } from '../helpers/utils/confirm-tx.util';
+import { getTokenFiatAmount } from '../helpers/utils/token-util';
+import { isEqualCaseInsensitive } from '../../shared/modules/string-utils';
+import { useTokenTracker } from './useTokenTracker';
+
+export const useAccountTotalFiatBalance = (
```

Comment:
`ui/components/app/asset-list/asset-list.test.js` passing proves that moving that logic to here works flawlessly!